Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthExceptionUtil.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2021 JetBrains s.r.o.
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azuretools.authmanage;
+
+import com.microsoft.azuretools.adauth.AuthError;
+import com.microsoft.azuretools.adauth.AuthException;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+@SuppressWarnings("unused")
+public final class AuthExceptionUtil {
+
+    /**
+     * When a Throwable is an AuthException that requires the user to re-authenticate,
+     * such as the one described in https://github.com/JetBrains/azure-tools-for-intellij/issues/438,
+     * returns the AuthException.
+     *
+     * @param throwable Throwable to check.
+     * @return When a Throwable is an AuthException that requires the user to re-authenticate, returns the AuthException. Returns null otherwise.
+     */
+    public static AuthException GetUserRetryableAuthException(Throwable throwable) {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.